### PR TITLE
Opto matrix analyzer

### PR DESCRIPTION
**Pull request by meganbkratz**
A new branch addressing issues brought up in PR #135. (Mostly took out changes to db schemas)

 -- added ability to specify different pre- and post-synaptic cell groups
-- change group_results function so that a pair can be included in the aggregate for multiple pair_groups if it fits in multiple pair_groups
-- added ext_id to experiment table

### CODE REVIEW

File: multipatch_analysis/matrix_analyzer/matrix_display.py [hunk 1]
```diff
@@ -216,7 +219,10 @@ def update_matrix_display(self, results, group_results, cell_classes, cell_group
         # attempt to fix so it doesn't break in mp_a\ui\graphics.py
         # at line 90. 
         rows = []
-        for i,cell_class in enumerate(cell_classes):
+        cols = []
+        pre_names = [pre.name for pre in pre_cell_classes] ## equality comparison happens on names, so need to convert to list of names here to ask if cell is in the list later
+        post_names = [post.name for post in post_cell_classes]
+        for i,cell_class in enumerate(set(pre_cell_classes+post_cell_classes)):
```
Comment (scseeman) [inline multipatch_analysis/matrix_analyzer/matrix_display.py]: according to Luke this randomizes the pre/post cell_classes and may be the cause of the unorganized matrix labels

**Comment by campagnola** [comment]
The online conflict resolution tool seems to be b0rk; but here's how I mashed together the cell group ui changes:
```
        self.cell_class_groups = cell_class_groups
        combo_def = {'name': 'pre/post', 'type':'list', 'value':'both', 'values':['both', 'presynaptic', 'postsynaptic']}
        cell_group_list = [{'name': group, 'type': 'bool', 'children':[combo_def], 'expanded':False} for group in self.cell_class_groups.keys()]
        layer = [{'name': 'Define layer by:', 'type': 'list', 'values': ['target layer', 'annotated layer'], 'value': 'target layer'}]
        children = layer + cell_group_list
        self.params = Parameter.create(name="Cell Classes", type="group", children=children)
        for p in self.params.children():
            p.sigValueChanged.connect(self.expand_param)
```

**Comment by campagnola** [comment]
Also, I just tested this and adding `experiment.ext_id` breaks the MP pipeline; can you remove that too? Sorry :(

**Comment by campagnola** [comment]
One other thing I had to do in config.py:
```
if hasattr(yaml, 'FullLoader'):
    # pyyaml new API
    config = yaml.load(open(configfile, 'rb'), Loader=yaml.FullLoader)
else:
    # pyyaml old API
    config = yaml.load(open(configfile, 'rb'))
```

**Pull request merged**